### PR TITLE
Fix FSM bug: conflicts on unilateral transactions

### LINKED ISSUE #3155: SC FSM: conflicts on unilateral transactions
## Expected Behavior
UserA can not interfere with UserB's unilateral transactions

## Actual Behavior

If UserB asks its FSM to produce an unilateral transaction for her, if meanwhile UserA produces an off-chain update a conflict is detected and both transactions are discarded.

### PR DESCRIPTION
**Pull request by velzevur**
Fixes #3155 

This PR had been produced with the support of the Aeternity Foundation. You can read about this in [the forum](https://forum.aeternity.com/t/active-aeternity-node-maintenance-iris-hard-fork-release-candidate/7365/).

### CODE REVIEW

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 1]
```diff
@@ -5776,3 +5797,124 @@ missing_malicious_snapshot_and_fp_based_on_it(Cfg0) ->
...
     ok.
 
+close_solo_and_conflict(Cfg) ->
+    Debug = get_debug(Cfg),
+    #{ i := #{fsm := FsmI} = I
+     , r := #{fsm := FsmR} = R
+     , spec := #{ initiator := PubI
+                , responder := PubR } = Spec} = create_channel_([?SLOGAN|Cfg]),
+    ok = rpc(dev1, aesc_fsm, close_solo, [FsmI, #{}]),
+    rpc(dev1, aesc_fsm, upd_transfer, [FsmR, PubR, PubI, 2]),
```
Comment (gorbak25) [inline apps/aechannel/test/aesc_fsm_SUITE.erl:5807]: Could you match that this returns ok? Later you assume that you get a signing request for the responder so this must return ok

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 2]
```diff
@@ -5776,3 +5797,124 @@ missing_malicious_snapshot_and_fp_based_on_it(Cfg0) ->
...
+
+slash_and_conflict(Cfg) ->
+    Debug = get_debug(Cfg),
+    #{ i := #{fsm := FsmI} = I
+     , r := #{fsm := FsmR} = R
+     , spec := #{ initiator := PubI
+                , responder := PubR } = Spec} = create_channel_([?SLOGAN|Cfg]),
+    {_I1, _R1, _SlashTx} = prepare_for_slashing(I, R, Spec, Cfg),
+    ok = rpc(dev1, aesc_fsm, slash, [FsmI, #{}]),
+    rpc(dev1, aesc_fsm, upd_transfer, [FsmR, PubR, PubI, 2]),
```
Comment (gorbak25) [inline apps/aechannel/test/aesc_fsm_SUITE.erl:5840]: same as above

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 3]
```diff
@@ -5776,3 +5797,124 @@ missing_malicious_snapshot_and_fp_based_on_it(Cfg0) ->
...
+snapshot_and_conflict(Cfg) ->
+    Debug = get_debug(Cfg),
+    #{ i := #{fsm := FsmI} = I
+     , r := #{fsm := FsmR} = R
+     , spec := #{ initiator := PubI
+                , responder := PubR }} = create_channel_([?SLOGAN|Cfg]),
+    {_I1, _R1} = do_update(PubI, PubR, 2, I, R, Debug, Cfg),
+    ok = rpc(dev1, aesc_fsm, snapshot_solo, [FsmI, #{}]),
+    {ok, Round} = rpc(dev1, aesc_fsm, get_round, [FsmI]),
+    rpc(dev1, aesc_fsm, upd_transfer, [FsmR, PubR, PubI, 2]),
```
Comment (gorbak25) [inline apps/aechannel/test/aesc_fsm_SUITE.erl:5863]: same as above

**Comment by velzevur** [comment]
Addressed comments and rebased on recent master

**Review by gorbak25 — APPROVED** [review]

**Pull request merged**